cloud/pkg/admissioncontroller/admission.go

```diff
@@ -222,31 +223,49 @@ func (ac *AdmissionController) registerWebhooks(opt *options.AdmissionOptions, c
 		},
 	}
 
-	return registerValidateWebhook(ac.Client.AdmissionregistrationV1beta1().ValidatingWebhookConfigurations(),
-		[]admissionregistrationv1beta1.ValidatingWebhookConfiguration{deviceModelCRDWebhook})
-}
+	objectSelector, err := metav1.ParseToLabelSelector("app-role.kubernetes.io=edge")
+	if err != nil {
+		return err
+	}
+	offlineMigrationWebhook := admissionregistrationv1beta1.MutatingWebhookConfiguration{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: OfflineMigrationConfigName,
+		},
+		Webhooks: []admissionregistrationv1beta1.MutatingWebhook{
+			{
+				Name:           OfflineMigrationWebhookName,
+				ObjectSelector: objectSelector,
+				Rules: []admissionregistrationv1beta1.RuleWithOperations{{
+					Operations: []admissionregistrationv1beta1.OperationType{
+						admissionregistrationv1beta1.Create,
+						admissionregistrationv1beta1.Update,
+					},
+					Rule: admissionregistrationv1beta1.Rule{
+						APIGroups:   []string{"apps", "extensions", ""},
+						APIVersions: []string{"v1", "v1beta1"},
+						Resources:   []string{"deployments", "daemonsets", "statefulsets"},
```

Comment:
should detect the labels on pods not workloads
```suggestion
						APIGroups:   []string{""},
						APIVersions: []string{"v1"},
						Resources:   []string{"pods"},
```